azavea/azavea-branding-guide
# Add copy-paste functionality to markdown blocks

## Overview
This ended up being pretty difficult. Basically `_includes` components only want a string, but we were trying to pass a markdown block. Now the copy/paste functionality works, but I also had to update the version of clipboard.js that we were using, because I needed to perform the copy function using js rather than an html attribute.

- Use up-to-date version of clipboard plugin
- Use liquid to deliver content to include copy-paste.html components
- Update readme

### Demo
![copy-paste-email](https://user-images.githubusercontent.com/5672295/44232358-c8ac2700-a16e-11e8-9b31-b13adc73429f.gif)

### Notes
- For some reason I couldn't copy/paste the 2.0.0 minified js plugin into our file... I was getting weird errors. So I'm pulling it in using a CDN instead.
- Nice addition as a result of this change: the text gets highlighted when you copy paste!
- I was able to "capture" the markdown block in a variable, and then run `| markdown` on that variable. It works... but it's a little ugly in the code. Open to ideas for making this better.

## Testing Instructions
* `git pull`
* Go to Communication > Email Signatures
* Click a copy button on one of the email signatures to see that it works. 
* Go to Identity > Company
* Click a copy button to see that it works.

Closes #89
